In [2]:
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer


from pymatgen import Structure
from pymatgen import symmetry
from pymatgen.ext.matproj import MPRester
from pymatgen.io.cif import CifWriter
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.core.operations import SymmOp
from IPython.display import Image
with MPRester("izD7mJmnjhUOKyWGtZ") as m:

    # Structure for material id
    structure = m.get_structure_by_material_id("mp-30847")
    w = CifWriter(structure)
    w.write_file('mp-30847.cif')

    struct= SpacegroupAnalyzer(structure)
    new_s=SpacegroupAnalyzer(structure).get_primitive_standard_structure() #Trouve

    #ReseauDirect
    ResD= new_s.lattice

    a= ResD.a
    b= ResD.b
    c= ResD.c
    alpha= ResD.alpha
    beta= ResD.beta
    gamma= ResD.gamma

    #ReseauReciproque
    rec= ResD.reciprocal_lattice
    arec= rec.a
    brec= rec.b
    crec= rec.c
    alpharec= rec.alpha
    betarec= rec.beta
    gammarec= rec.gamma

    #Type de maille, cristal, group
    maille= struct.get_lattice_type()
    cristal= struct.get_crystal_system()
    GrouP= struct.get_point_group_symbol()

    print("Vecteurs de base du réseau direct:")
    print(ResD)
    print("")
    print("Longeur de A: %f" %(a))
    print("Longeur de B: %f" %(b))
    print("Longeur de C: %f" %(c))

    print("Angle alpha: %f" %(alpha))
    print("Angle beta: %f" %(beta))
    print("Angle gamma: %f" %(gamma))
    print("Vecteurs de base du réseau réciproque:")
    print(rec)
    print("")
    print("Longeur du vecteur A*: %f" %(arec))
    print("Longeur du vecteur B*: %f" %(brec))
    print("Longeur du vecteur C*: %f" %(crec))
    print("Angle alpha*: %f" %(alpharec))
    print("Angle beta*: %f" %(betarec))
    print("Angle gamma*: %f" %(gammarec))
    print("Type de maille : " + maille)
    print("")
    print("Système cristallin : " + cristal)
    print("")
    print("Groupe ponctuel : " + GrouP)


Vecteurs de base du réseau direct:
0.000000 3.115911 3.115911
3.115911 0.000000 3.115911
3.115911 3.115911 0.000000

Longeur de A: 4.406564
Longeur de B: 4.406564
Longeur de C: 4.406564
Angle alpha: 60.000000
Angle beta: 60.000000
Angle gamma: 60.000000
Vecteurs de base du réseau réciproque:
-1.008242 1.008242 1.008242
1.008242 -1.008242 1.008242
1.008242 1.008242 -1.008242

Longeur du vecteur A*: 1.746327
Longeur du vecteur B*: 1.746327
Longeur du vecteur C*: 1.746327
Angle alpha*: 109.471221
Angle beta*: 109.471221
Angle gamma*: 109.471221
Type de maille : cubic

Système cristallin : cubic

Groupe ponctuel : -43m


# Vecteurs du réseau direct
On trouve comme vecteurs de base : <br />
$\overrightarrow{a}$ : (0.000000 3.115911 3.115911)  <br />
$\overrightarrow{b}$ : (3.115911 0.000000 3.115911) <br />
$\overrightarrow{c}$ : (3.115911 3.115911 0.000000)<br /> 


Si on pose L=3.115911 Å <br />
On trouve alors :<br />
$
\overrightarrow{a} : L(0 1 1) \\
\overrightarrow{b} : L(1 0 1) \\
\overrightarrow{c} : L(1 1 0) \\ $


# Vecteurs du réseau réciproque
On trouve comme vecteurs du réseau réciproque : <br />
$\overrightarrow{a*}$ : (-1.008242 1.008242 1.008242)  <br />
$\overrightarrow{b*}$ : (1.008242 -1.008242 1.008242) <br />
$\overrightarrow{c*}$ : (1.008242 1.008242 -1.008242)<br /> 
Si on pose L*=1.008242 Å <br />
On trouve alors :<br />
$\overrightarrow{a}$ : L(-1 1 1) <br />
$\overrightarrow{b}$ : L(1 -1 1) <br />
$\overrightarrow{c}$ : L(1 1 -1) <br />



## Type de maille : Cubique
## Système Cristallin : Cubique
## Groupe Ponctuel : -43m